src/RA_Dlg_MemBookmark.h

```diff
@@ -51,56 +101,62 @@ class Dlg_MemBookmark
-    INT_PTR MemBookmarkDialogProc(HWND, UINT, WPARAM, LPARAM);
-
     void InstallHWND(HWND hWnd) noexcept { m_hMemBookmarkDialog = hWnd; }
-    HWND GetHWND() const noexcept { return m_hMemBookmarkDialog; }
+    inline auto GetHWND() const noexcept { return m_hMemBookmarkDialog; }
     BOOL IsActive() const noexcept;
 
-    std::vector<MemBookmark*> Bookmarks() { return m_vBookmarks; }
+    _NODISCARD inline auto& Bookmarks() noexcept { return m_vBookmarks; }
+    _NODISCARD inline auto& Bookmarks() const noexcept { return m_vBookmarks; }
     void UpdateBookmarks(bool bForceWrite);
-    void AddBookmark(MemBookmark* newBookmark) { m_vBookmarks.push_back(newBookmark); }
+    inline auto AddBookmark(MemBookmark&& newBookmark)
+    {
+        m_vBookmarks.push_back(std::move(newBookmark));
+        Sort(); // sort after adding
+    }
     void WriteFrozenValue(const MemBookmark& Bookmark);
 
     void OnLoad_NewRom();
 
-private:
-    int m_nNumOccupiedRows;
+    static INT_PTR CALLBACK s_MemBookmarkDialogProc(HWND, UINT, WPARAM, LPARAM);
 
+protected:
+    _NODISCARD INT_PTR MemBookmarkDialogProc(HWND, UINT, WPARAM, LPARAM);
+
+private:
     void PopulateList();
     void SetupColumns(HWND hList);
     BOOL EditLabel(int nItem, int nSubItem);
     void AddAddress();
-    void ClearAllBookmarks();
+    void ClearAllBookmarks() noexcept;
     unsigned int GetMemory(unsigned int nAddr, int type);
 
     void ExportJSON();
-    void ImportFromFile(std::wstring filename);
+    void ImportFromFile(std::wstring&& filename);
     void GenerateResizes(HWND hDlg);
     std::wstring ImportDialog();
 
-    void AddBookmarkMap(_In_ const MemBookmark* bookmark)
+public:
+    auto BookmarkExists(const ra::ByteAddress& nAddr) const
     {
-        if (m_BookmarkMap.find(bookmark->Address()) == m_BookmarkMap.end())
-            m_BookmarkMap.insert(std::map<ra::ByteAddress, std::vector<const MemBookmark*>>::value_type(bookmark->Address(), std::vector<const MemBookmark*>()));
-
-        std::vector<const MemBookmark*> *v = &m_BookmarkMap[bookmark->Address()];
-        v->push_back(bookmark);
+        // Should be sorted
+        return (std::binary_search(m_vBookmarks.cbegin(), m_vBookmarks.cend(), nAddr));
     }
 
-public:
-    const MemBookmark* FindBookmark(const ra::ByteAddress& nAddr) const
+    auto& FindBookmark(const ra::ByteAddress& nAddr) const
     {
-        std::map<ra::ByteAddress, std::vector<const MemBookmark*>>::const_iterator iter = m_BookmarkMap.find(nAddr);
-        return(iter != m_BookmarkMap.end() && iter->second.size() > 0) ? iter->second.back() : nullptr;
+        static const MemBookmark dummyBookmark;
+        if (BookmarkExists(nAddr)) // Should be sorted when it was added (if it exists)
+            // TBD: There could be a better algorithm but this should be OK
+            return *std::find(m_vBookmarks.cbegin(), m_vBookmarks.cend(), nAddr);
+        return dummyBookmark; // throw?
```

Comment:
This basically does what was here before but uses a vector instead. A bookmark with an address of 0 is considered invalid but wasn't sure if we should've thrown instead.

Reply:
You could theorhetically bookmark address 0, it shows up in the inspector. Prefer using MAX_INT instead.